### ISSUE TITLE
Rule Change: Enhance `no-self-assign`

### ISSUE THREAD

**Issue by erosman** [eslint/eslint#19444]
### What rule do you want to change?

no-self-assign

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

Other

### Example code

There are occasions, especially in DOM, that self-assignment is common practice. While it is currently possible to apply `{"props": false}`, a blanket override might not be desirable.

A common DOM self-assignment is the following method to replace content of an element with its plain text.

```js
element.textContent = element.textContent;
```

It would be beneficial to exclude DOM properties (or some of) by default in `no-self-assign`, or add an option specifically for this purpose.

### What does the rule currently do for this code?

Generates an error

### What will the rule do after it's changed?

Not generate an error

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by nzakas** [comment]
Thanks for the feedback. What you're describing is an edge case, so it's best to use a disable comment around that code if you want to omit that check.
